src/components/Todo/Todo.js

```diff
@@ -1,122 +1,64 @@
     }
 
-
     handleChange = (event) => {
        this.setState({
              value : event.target.value,
        })
 
     }
-
     onAddTodoHandler = (event , inputValue) => {
       event.preventDefault();
-      const newArr = [...this.state.todoList];
       const todoData = {
-        name : inputValue ,
+        name : inputValue.trim() ,
         Date : new Date().toDateString(),
         checked : false ,
       }
-      axios.post("/ListTodo.json" , todoData)
-            .then(response => {
-              newArr.push({...todoData ,id:response.data.name})
-              this.setState({
-                todoList : newArr ,
-                value : ''
-              })
-            }).catch(err => {
-              console.log(err)
-            })
+      this.props.onAddTodo(todoData)
     }
-    onRemoveTodoTasks = index => {
-        axios.delete(`/ListTodo/${index}.json`)
-              .then(res => {
-                this.setState( previousState => {
-                  return {
-                    todoList : previousState.todoList.filter(item => item.id !== index)
-                  };
-                });
-              }).catch(err=>{
-                console.log(err);
-              })
-       
-    }
-
-    onCheckBtnHandler = (index , item) => {
-      const updateChecked = {
-          ...item,
-          checked : !item.checked
-      }
-      axios.put(`/ListTodo/${item.id}.json`,updateChecked)
-             .then(res => {
-               let newArr = [...this.state.todoList.filter(item => index !== item.id)];
-               newArr.push({...updateChecked})
-               this.setState({
-                  todoList : newArr,
-                })
-             }).catch(err => {
-
-             })
-
-    }
-
     render() {
-
-
+      let redirectWhenFails= null
+      if(this.props.error) {
+        redirectWhenFails= <Redirect to = {routsPath.ROOT_PATH}  />
+      }
       let todoListData = (
           <p style={{alignItems:'center',color:'#6200EE'}}>Let's Add SOME TO-DO</p>
       )
-      if(this.state.todoList != null) {
+      if(this.props.todoList != null) {
         todoListData = (
             <List className={classes.List}>
-                        {this.state.todoList.map((itemTodo) => (
+                        {this.props.todoList.map((itemTodo) => (
 
                               <ListTodo key ={itemTodo.id} 
                                         checked={itemTodo.checked}
                                         todoText={itemTodo.name}
-                                        clickToRemove={()=>this.onRemoveTodoTasks(itemTodo.id)}
-                                        checkBtn ={()=> this.onCheckBtnHandler(itemTodo.id,itemTodo)}/>
+                                        clickToRemove={()=>this.props.onRemoveTodo(itemTodo.id)}
+                                        checkBtn ={()=> this.props.onCheckBtnHandler(itemTodo)}/>
                         ))}
             </List> 
```

Comment:
Why changing the state of one item changes the order of the list?

Reply:
because I filtrate the object and then contact it, I try a different approach it's doesn't work correctly (it needs 2 clicks to update Ui check I am confused but when I click the first time I make console.log for the state its update but Ui doesn't)

Reply:
Solved